src/wallet/rpcwallet.cpp

```diff
@@ -3007,6 +3011,112 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
     }
 }
 
+UniValue z_getnewaccount(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+    if (fHelp || params.size() > 0)
+        throw runtime_error(
+            "z_getnewaccount\n"
+            "\nSelects the next available sequential account number within the UA-compatible HD seed phrase,"
+            "\ngenerates the corresponding spend authorities for the best and second-best shielded pools, and the transparent pool,"
+            "\nreturns the new account number and the default / first UA containing the default addresses for each pool.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"account\": n,       (numeric) next available account number\n"
+            "  \"unifiedaddress\"    (string) The default address for this account\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("z_getnewaccount", "")
+            + HelpExampleRpc("z_getnewaccount", "")
+        );
+
+    if (!fExperimentalOrchardWallet) {
+        throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: the Orchard wallet experimental extensions are disabled.");
+    }
+    int64_t account = 999999; // TODO placeholder
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("account", account);
+    result.pushKV("unifiedaddress", "TODO");
+    return result;
+}
+
+UniValue z_getaddressforaccount(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+    if (fHelp || params.size() < 1 || params.size() > 3)
+        throw runtime_error(
+            "z_getaddressforaccount account ( diversifier_index [\"protocol\", ...] )\n"
+            "\nFor the given account number, return a UA in accordance with the remaining arguments."
+            "\nIf no diversifier index is given (or the string \"*\"), the next unused index will be selected."
+            "\nCompute a Unified Address corresponding to the diversifier index, containing addresses"
+            "\nfor the given protocols (default all).\n"
```

Comment:
I don't think we should promise that the default is all protocols. That may be reasonable while we only have transparent, Sapling, and Orchard protocols, but I can see us wanting to change that if either we add another protocol, or a security issue in an existing protocol is found.